src/osemgrep/language_server/Test_LS_e2e.ml

```diff
@@ -152,6 +153,14 @@ let open_and_write_default_content ?(mode = []) file =
    So, when running in JSCaml, let's just not use pauses. *)
 let lwt_pause () = if !Common.jsoo then Lwt.return_unit else Lwt.pause ()
 
+let with_timeout (f : unit -> 'a Lwt.t) : unit -> 'a Lwt.t =
```

Comment:
Should a (slightly more generic) version of this just be in `Lwt_platform`?

Reply:
we could do a with_sleep